### PR TITLE
chores: card update

### PR DESCRIPTION
**Pull request by SzDabrowski**

### CODE REVIEW

**Review by NerdbordMentor — CHANGES_REQUESTED** [review]
## Code Review Report

### Task goal
Implement editing existing flashcards

### Requirements
- It should be able to turn on `Edit` mode when clicking on a pencil icon
- It should be able to `Cancel` editing
- It should be able to `Save` editing

### Code Review

#### src/App.tsx
The changes in this file seem to meet the requirements. The `updateCard` function has been introduced to handle the card updates. However, there is a commented line `//setCardsData();` which should be uncommented and used to update the state of the cards data after editing.

#### src/components/Card/BackCardEdit.tsx
The changes in this file seem to meet the requirements. The `updateCard` function is used to save the changes made to the card. The `cancelPageClick` function is used to cancel the editing. The `editMode` state is toggled when the pencil icon is clicked.

#### src/components/Card/Card.tsx
The changes in this file seem to meet the requirements. The `editMode` state is toggled when the pencil icon is clicked. The `updateCard` function is used to save the changes made to the card.

#### src/components/Card/FrontCardEdit.tsx
The changes in this file seem to meet the requirements. The `updateCard` function is used to save the changes made to the card. The `cancelPageClick` function is used to cancel the editing. 

### Conclusion
The code changes cover all the task requirements. However, there are some minor issues that need to be addressed, such as uncommenting the line `//setCardsData();` in `src/App.tsx` file, which is crucial for updating the state of the cards data after editing.

**Review by NerdbordMentor — CHANGES_REQUESTED** [review]
## Code Review Report

### Task goal
Implement editing existing flashcards

### Requirements
- It should be able to turn on `Edit` mode when clicking on a pencil icon
- It should be able to `Cancel` editing
- It should be able to `Save` editing

### Code Review

#### src/App.tsx
The changes in this file seem to cover the task requirements. The `updateCard` function has been added to handle the editing of existing flashcards. However, there is a commented line `//setCardsData();` which should be uncommented and used to update the state of the cards data after editing.

#### src/components/Card/BackCardEdit.tsx
The changes in this file also cover the task requirements. The `updateCard` function has been added to handle the saving of the edits. The `Cancel` button now correctly toggles the edit mode off.

#### src/components/Card/Card.tsx
The changes in this file cover the task requirements. The `editMode` state has been added and is toggled when the pencil icon is clicked, which turns on the `Edit` mode. The `updateCard` function has been added to handle the saving of the edits.

#### src/components/Card/FrontCardEdit.tsx
The changes in this file cover the task requirements. The `updateCard` function has been added to handle the saving of the edits. The `Cancel` button now correctly toggles the edit mode off.

### Conclusion
The code changes cover the task requirements. However, there is a minor issue in `src/App.tsx` where the updated cards data is not being set in the state after editing. This should be fixed for the changes to take effect.

File: src/App.tsx [hunk 1]
```diff
@@ -67,6 +64,17 @@ function App() {
     setDisplayNewCard(false);
   };
 
+  const updateCard = (id: number, cardSide: string, input: string) => {
+    const updateCard = cardsData.map((card) => {
+      if (card.id === id) {
+        return { ...card, cardSide: input }
```
Comment (Sumick) [inline src/App.tsx:70]: **issue:** Logic is correct but you have bug here, as you pass `cardSide` incorrectly. You should close it in `[]` to have dynamic key.

```js
 const updateCard = (id: number, cardSide: string, input: string) => {
    const updateCard = cardsData.map((card) => {
      if (card.id === id) {
        return { ...card, [cardSide]: input }
      }
      return card
    });

    setCardsData(updateCard);
    }
```

**Review by Sumick — COMMENTED** [review]
I think I found a bug in your function, let me know if that was the problem. Anyway, we should not merge unreviewed or rejected pulls, as you may introduce bugs into production environment.

**Pull request merged**